eclipse-collections/src/main/java/org/eclipse/collections/impl/list/mutable/MutableListFactoryImpl.java

```diff
@@ -14,9 +14,9 @@
 import org.eclipse.collections.api.factory.list.MutableListFactory;
 import org.eclipse.collections.api.list.MutableList;
 
-public enum MutableListFactoryImpl implements MutableListFactory
+public class MutableListFactoryImpl implements MutableListFactory
 {
-    INSTANCE;
+    public static final MutableListFactory INSTANCE = new MutableListFactoryImpl();
```

Comment:
I kept a constant named `INSTANCE` to improve backwards compatibility. However, this class is no longer a singleton. At runtime, there will really be two instances of MutableListFactoryImpl. The one in api.Lists will be instantiated by the ServiceLoader, which instantiates the class with the zero-arg constructor. The one in impl.Lists will be this INSTANCE. And of course, it would be possible for users to construct more.

I thought about sharing an instance like this:
```java
public final class Lists
{
    public static final ImmutableListFactory immutable = org.eclipse.collections.api.factory.list.Lists.immutable;
    public static final MutableListFactory mutable = org.eclipse.collections.api.factory.list.Lists.mutable;
    public static final FixedSizeListFactory fixedSize = org.eclipse.collections.api.factory.list.Lists.fixedSize;
    public static final MutableListFactory multiReader = MultiReaderMutableListFactory.INSTANCE;
}
```
However, two instances feels simpler, and avoids rare but strange edge cases like confusion around two copies of the same jar on the classpath.